added section about parallelism to delegation (#74158)
ansible/ansible: docs/docsite/rst/user_guide/playbooks_delegation.rst

            body: "{{ mail_body }}"
          run_once: True

.. note::

>     - The `ansible_host` variable and other connection variables, if present, will reflect information of the host a task is delegated to, not the inventory_hostname.

Reviewer: ```suggestion
    - The `ansible_host` variable and other connection variables, if present, will reflect information about the host a task is delegated to, not the inventory_hostname.
```

>     - The `ansible_host` variable and other connection variables, if present, will reflect information of the host a task is delegated to, not the inventory_hostname.

Reviewer 2: Per our style guide, avoid future tense.

```suggestion
    - The `ansible_host` variable and other connection variables, if present, reflects information of the host a task is delegated to, not the inventory_hostname.
```

.. _delegate_parallel:

> Parallelism
> -----------

Reviewer 2: ```suggestion
Delegation and parallel execution
---------------------------------
```

> By default Ansbile tasks are executed in parallel, delegating a task does not change this and does not handle concurrency issues (mulitple forks writing to same file).

Reviewer 3: ```suggestion
By default Ansible tasks are executed in parallel. Delegating a task does not change this and does not handle concurrency issues (multiple forks writing to same file).
```

> By default Ansible tasks are executed in parallel. Delegating a task does not change this and does not handle concurrency issues (multiple forks writing to same file).

Reviewer: ```suggestion
By default Ansible tasks are executed in parallel. Delegating a task does not change this and does not handle concurrency issues (multiple forks writing to the same file).
```

> Most commonly users are affected by this when updating a single file on a single delegated to host for all hosts (via copy/template/lineinfile/etc), they will still operate in parallel forks (default 5) and overwrite each other.

Reviewer: ```suggestion
Most commonly, users are affected by this when updating a single file on a single delegated to host for all hosts (using the ``copy``, ``template``, or ``lineinfile`` modules, for example). They will still operate in parallel forks (default 5) and overwrite each other.
```

This can be handled in several ways::

>     - name: "handle concurrency via a loop on the hosts with `run_once: true`"

Reviewer: ```suggestion
    - name: "handle concurrency with a loop on the hosts with `run_once: true`"
```

      lineinfile: "<options here>"
      run_once: true
      loop: '{{ ansible_play_hosts_all }}'

> By using an intermidate play with  `serial: 1` or using  `throttle: 1` at task level.

Reviewer 3: ```suggestion
By using an intermediate play with  `serial: 1` or using  `throttle: 1` at task level.
```

> By using an intermediate play with  `serial: 1` or using  `throttle: 1` at task level.

Reviewer 2: Another example would be good here, and/or link to https://docs.ansible.com/ansible/latest/user_guide/playbooks_strategies.html#setting-the-batch-size-with-serial and https://docs.ansible.com/ansible/latest/user_guide/playbooks_strategies.html#restricting-execution-with-throttle.

            body: "{{ mail_body }}"
          run_once: True

.. note::
    - The `ansible_host` variable and other connection variables, if present, will reflect information of the host a task is delegated to, not the inventory_hostname.

.. _delegate_parallel:

Parallelism
-----------
By default Ansbile tasks are executed in parallel, delegating a task does not change this and does not handle concurrency issues (mulitple forks writing to same file).
Most commonly users are affected by this when updating a single file on a single delegated to host for all hosts (via copy/template/lineinfile/etc), they will still operate in parallel forks (default 5) and overwrite each other.

This can be handled in several ways::

    ---
    - name: handle concurrency via a loop on the hosts with `run_once: true`:
      lineinfile: ...
      run_once: true

>       loop: '{{ q("inventory_hostnames" ...}}'

Reviewer 4: The example block needs to be valid YAML + Jinja2 syntax:

```suggestion
      loop: '{{ q("inventory_hostnames") }}'
```
